internal/operators/ocs/validations.go

```diff
@@ -0,0 +1,209 @@
+	OCSRequiredHosts               int64 `envconfig:"OCS_MINIMUM_HOST" default:"3"`
+	OCSRequiredCPUCount            int64 `envconfig:"OCS_REQUIRED_CPU_COUNT" default:"30"`
+	OCSRequiredRAMGB               int64 `envconfig:"OCS_REQUIRED_RAM_GB" default:"72"`
+	OCSRequiredCompactModeCPUCount int64 `envconfig:"OCS_REQUIRED_COMAPCT_MODE_CPU_COUNT" default:"12"` // compact mode cpu requirements ( 4(master)+8(OCS) cpus)
+	OCSRequiredCompactModeRAMGB    int64 `envconfig:"OCS_REQUIRED_COMAPCT_MODE_RAM_GB" default:"96"`    //compact mode ram requirements (24(master)+72(OCS) GB)
+	OCSMasterWorkerHosts           int   `envconfig:"OCS_REQUIRED_MASTER_WORKER_HOSTS" default:"5"`
+	OCSMinimalDeployment           bool  `envconfig:"OCS_MINIMAL_DEPLOYMENT" default:"false"`
+	OCSDisksAvailable              int64 `envconfig:"OCS_DISKS_AVAILABLE" default:"3"`
+}
+
+func SetOperatorStatus(cluster *models.Cluster, status string) error {
+	var operators models.Operators
+	err := json.Unmarshal([]byte(cluster.Operators), &operators)
+	if err != nil {
+		return err
+	}
+	for _, operator := range operators {
+		if operator.OperatorType == models.OperatorTypeOcs {
+			operator.Status = status
+			break
+		}
+	}
+	clusterOperators, err := json.Marshal(operators)
+	if err != nil {
+		return err
+	}
+
+	cluster.Operators = string(clusterOperators)
+	return nil
+}
+
+// ValidateOCSRequirements is used to validate min requirements of OCS
+func (o *ocsValidator) ValidateOCSRequirements(cluster *models.Cluster) bool {
+	var status string
+	hosts := cluster.Hosts
+
+	if int64(len(hosts)) < o.OCSRequiredHosts {
+		status = "Insufficient hosts to deploy OCS. A minimum of 3 hosts is required to deploy OCS. "
+		err := SetOperatorStatus(cluster, status)
+		o.log.Info("Setting Operator Status")
+		if err != nil {
+			o.log.Fatal("Error occured while Marshalling or Unmarshaling Operators ", err)
+			return false
+		}
+		o.log.Info(status)
+		return false
+	}
+	var CPUCount int64 = 0                       //count the total CPUs on the cluster
+	var TotalRAM int64 = 0                       // count the total available RAM on the cluster
+	var DiskCount int64 = 0                      // count the total disks on all the hosts
+	var HostsWithDisks int64 = 0                 // to determine total number of hosts with disks. OCS requires atleast 3 hosts with non-bootable disks                           // to claim for the PV created by LocalVolumeSet
+	if int64(len(hosts)) == o.OCSRequiredHosts { // for only 3 hosts, we need to install OCS in compact mode
+		for _, host := range hosts {
+			err := o.nodeRequirements(host, &CPUCount, &TotalRAM, &DiskCount, &HostsWithDisks)
+			if err != nil {
+				o.log.Fatal("Error occured while calculating Node requirements ", err)
+				return false
+			}
+		}
+
+	} else if len(hosts) == o.OCSMasterWorkerHosts { // not supporting OCS installation for 2 Workers and 1 Master
```

Comment:
The numbers in the comment are incorrect

Reply:
A better comparison would be <= o.OCSMasterWorkerHosts (just to future proof it as 3 masters + 1 worker is not supported in OCP atm)

Reply:
fixed